include/opendht/indexation/pht.h

```diff
@@ -166,6 +176,125 @@ class Pht {
     void insert(Key k, Value v, Dht::DoneCallbackSimple cb = {});
 
 private:
+    class Cache {
+    public:
+        /**
+         * Insert all needed node into the tree according to a prefix
+         * @param p : Prefix that we need to insert
+         */
+        void insert(const Prefix& p) {
+            size_t i = 0;
+            auto now = clock::now();
+
+            std::shared_ptr<Node> curr_node;
+
+            while ( ( leaves_.size() > 0 && leaves_.begin()->first + NODE_EXPIRE_TIME < now )
+                    || leaves_.size() > MAX_ELEMENT ) {
+
+                leaves_.erase(leaves_.begin());
+            }
+
+            if ( !(curr_node = root_.lock()) ) {
+
+                /* Root does not exist, need to create one*/
+                curr_node = std::make_shared<Node>();
+                root_ = curr_node;
+            }
+            
+            curr_node->last_reply = now;
+
+            /* Iterate through all bit of the Blob */
+            for ( i = 0; i < p.size_; i++ ) {
+
+                /* According to the bit define which node is the next one */
+                auto& next = ( p.isActivBit(i) ) ? curr_node->right_child : curr_node->left_child;
+
+                /** 
+                 * If lock, node exists
+                 * else create it
+                 */
+                if (auto n = next.lock()) {
+                    curr_node = std::move(n);
+                } else {
+                    /* Create the next node if doesn't exist*/
+                    auto tmp_curr_node = std::make_shared<Node>();
+                    tmp_curr_node->parent = curr_node;
+                    next = tmp_curr_node;
+                    curr_node = std::move(tmp_curr_node);
+                }
+
+                curr_node->last_reply = now;
+            }
+
+            /* Insert the leaf (curr_node) into the multimap */
+            leaves_.insert(std::pair<time_point, std::shared_ptr<Node>>(now, curr_node ));
```

Comment:
Here's some c++11 tips. You can do this:
```cpp
leaves_.insert(std::pair<time_point, std::shared_ptr<Node>>(now, curr_node));
```
but, you can shrink it to:
```cpp
leaves_.insert(std::make_pair(now, curr_node));
```
furthermore, you can do:
```cpp
leaves_.emplace(now, curr_node);
```
Finally, *if I'm not mistaken*, the leaf would be copied constructed from `curr_node`. Then, you can enhance the code by making `std::multimap::emplace` use the move constructor of `std::shared_ptr`:
```cpp
leaves_.emplace(now, std::move(curr_node));
```